### PR TITLE
feat: [CAL-910] Pre-select country for phone number (CALCOM-5759)

### LINKED ISSUE #5759: [CAL-910] Pre-select country for phone number
Currently when a person has to enter his phone number, he has to select the country or enter the code after "+"

image

We should add the possibility to pre-select a country for a form

image

<sub>[CAL-910](https://linear.app/calcom/issue/CAL-910/pre-select-country-for-phone-number)</sub>

### PR DESCRIPTION
**Pull request by gitstart-calcom**
Here is the GitStart Ticket for this pull request: [CALCOM-5759](https://developers.gitstart.com/client/calcom/ticket/CALCOM-5759)

- Please mark your PR as "ready for review" when you have completed all requirements.
- Please ensure that your PR is reviewed by at least 2 people after you have completed it.
- Comments for review or requested changes from the upstream client will be notified to you by email and will also show on the [ticket page](https://developers.gitstart.com/client/calcom/ticket/CALCOM-5759).
- After the PR is merged on the upstream, please merge this PR.
- If there exist merge conflicts on your current PR and you have to close the PR instead of merging it, please go to the [ticket page](https://developers.gitstart.com/client/calcom/ticket/CALCOM-5759) to manually finish the task that is related to this PR.

Happy hacking! 🎉

## Demo
https://www.loom.com/share/fa251854a45c4c339568db1045419063?sid=bcbb9309-825a-403f-8656-e414e7e1280a

### CODE REVIEW

File: packages/ui/form/PhoneInput.tsx [hunk 1]
```diff
@@ -59,17 +58,13 @@ function BasePhoneInput({ name, className = "", onChange, ...rest }: PhoneInputP
 
 const useDefaultCountry = () => {
   const [defaultCountry, setDefaultCountry] = useState("us");
```
Comment (gitstart-calcom) [inline packages/ui/form/PhoneInput.tsx:60]: We keep the initial state "us" if the country is not supported we keep displaying us when the page is loaded(as demonstrated on loom video)

File: packages/ui/form/PhoneInput.tsx [hunk 2]
```diff
@@ -59,17 +58,13 @@ function BasePhoneInput({ name, className = "", onChange, ...rest }: PhoneInputP
...
-      }
-    },
-  });
 
+  useEffect(() => {
+    const userCountryCode = navigator.language.split("-")[1];
+    if (userCountryCode && isSupportedCountry(userCountryCode)) {
+      setDefaultCountry(userCountryCode.toLowerCase());
+    }
+  }, []);
```
Comment (gitstart-calcom) [inline packages/ui/form/PhoneInput.tsx:67]: The old way to grab the "defaultCountry" was always returning "us", so I changed how we can grab the user location, now we use the language selected on navigator to display the PhoneInput with this country code initially

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
The route viewer.public.countryCode works fine when we deploy on vercel because we get the country code from this header  req?.headers?.["x-vercel-ip-country"]. you can check this event type https://cal.com/udit/udit-s-test-event to verify.

using navigator.language will not give correct result. I am getting UK(+44) whereas it should be india(+91) in the phone input

Screenshot 2023-06-16 at 5 00 00 PM

File: packages/ui/form/PhoneInput.tsx [hunk 3]
```diff
@@ -59,17 +58,13 @@ function BasePhoneInput({ name, className = "", onChange, ...rest }: PhoneInputP
 
 const useDefaultCountry = () => {
   const [defaultCountry, setDefaultCountry] = useState("us");
-  trpc.viewer.public.countryCode.useQuery(undefined, {
-    refetchOnWindowFocus: false,
-    refetchOnReconnect: false,
-    retry: false,
-    onSuccess: (data) => {
-      if (isSupportedCountry(data?.countryCode)) {
```
Comment (Udit-takkar) [inline packages/ui/form/PhoneInput.tsx:67]: we can add a condition here if data?.countryCode === "" (this happens when vercel headers are present with countryCode)  and then we use 
const userCountryCode = navigator.language.split("-")[1]

**Comment by gitstart-calcom** [comment]
Hi @Udit-takkar we made some changes, now we use axios (a lib that already existed in the project) to find the country code in case the way that already exists doesn't work.

File: packages/ui/form/PhoneInput.tsx [hunk 4]
```diff
@@ -58,14 +59,21 @@ function BasePhoneInput({ name, className = "", onChange, ...rest }: PhoneInputP
...
     retry: false,
-    onSuccess: (data) => {
-      if (isSupportedCountry(data?.countryCode)) {
-        setDefaultCountry(data.countryCode.toLowerCase());
+    onSuccess: async (data) => {
+      try {
+        if (isSupportedCountry(data?.countryCode)) {
+          setDefaultCountry(data.countryCode.toLowerCase());
+        } else {
+          const response = await axios.get("https://ipapi.co/json/");
```
Comment (Udit-takkar) [inline packages/ui/form/PhoneInput.tsx:72]: This is a paid API(https://ipapi.co/pricing/) not sure if we should use this or not? cc: @PeerRich @emrysal

Comment (gitstart-calcom) [reply packages/ui/form/PhoneInput.tsx:72]: We are using this api to get the location more precisely, but we can navigator.language instead as it's a secundary way to grab the countryCode

Comment (Udit-takkar) [reply packages/ui/form/PhoneInput.tsx:72]: @gitstart-calcom ok just use navigator.language for now.

**Review by Udit-takkar — APPROVED** [review]

**Pull request merged**